### PR TITLE
add RoleBindingRestriction e2e test

### PR DESCRIPTION
**Pull request by sallyom**
@enj  adding back test plus a few minor bits from [revert](https://github.com/openshift/origin/pull/23176)

### CODE REVIEW

**Comment by enj** [comment]
@sallyom see https://github.com/openshift/origin/pull/23176#pullrequestreview-250070356

**Comment by sallyom** [comment]
@enj whoops missed the review as to what to add back in.. fixing up now, thanks

File: test/extended/authorization/rolebinding_restrictions.go [hunk 1]
```diff
@@ -0,0 +1,226 @@
...
+				Kind:      authorizationapi.UserKind,
+				Namespace: ns,
+				Name:      "system:non-existing",
+			},
+		},
+		RoleRef: corev1.ObjectReference{Name: "role", Namespace: ns},
+	}
+}
+
+func roleBindingCreate(oc *exutil.CLI, useRBAC, shouldErr bool, ns, user, rb string) {
```
Comment (enj) [inline test/extended/authorization/rolebinding_restrictions.go]: I think this is a bit easier to read:

```go
func roleBindingCreate(oc *exutil.CLI, useRBAC, shouldErr bool, ns, user, rb string) {
	var rbrErr error
	switch {
	case useRBAC:
		_, rbrErr = oc.AdminKubeClient().RbacV1().RoleBindings(ns).Create(generateRbacUserRolebinding(ns, user, rb))
	default:
		_, rbrErr = oc.AdminAuthorizationClient().AuthorizationV1().RoleBindings(ns).Create(generateRolebinding(ns, user, rb))
	}
	if !shouldErr {
		o.Expect(rbrErr).NotTo(o.HaveOccurred())
		return
	}
	o.Expect(rbrErr).NotTo(BeNil())
	o.Expect(kerrors.IsForbidden(rbrErr)).To(o.BeTrue())
	expectedErrorString := fmt.Sprintf("rolebindings to User \"%s\" are not allowed in project", user)
	o.Expect(rbrErr.Error()).Should(o.ContainSubstring(expectedErrorString))
}
```

Comment (sallyom) [reply test/extended/authorization/rolebinding_restrictions.go]: yes, it is, thanks.  I didn't realize you reviewed this so I've pushed some cleanup to the e2e test, please will you take a look again at this file? thanks :slightly_smiling_face:

Comment (sallyom) [reply test/extended/authorization/rolebinding_restrictions.go]: also, I've made that change

**Comment by sallyom** [comment]
@enj had to clear the needs-rebase label, mind tagging again? thanks

**Pull request merged**